File: grimdb/sdk/php/src/Models/SyncStatus.php
<?php declare(strict_types=1);
namespace Grimlocker\Models;

class SyncStatus
{
    /**
     * @param SyncPeer[] $peers
     */
    public function __construct(
        public readonly array   $peers      = [],
        public readonly int     $lastSyncAt = 0,
        public readonly string  $deviceId   = '',
    ) {}

    public static function fromArray(array $data): self
    {
        $peers = array_map(
            fn(array $p) => SyncPeer::fromArray($p),
            $data['peers'] ?? [],
        );
        return new self(
            $peers,
            $data['last_sync_at'] ?? 0,
            $data['device_id']    ?? '',
        );
    }
}

File: grimdb/sdk/csharp/src/Models/SyncStatus.cs
using System.Text.Json.Serialization;

namespace Grimlocker.Models;

public record SyncPeer(
    [property: JsonPropertyName("device_id")] string DeviceId,
    [property: JsonPropertyName("host")]      string Host,
    [property: JsonPropertyName("port")]      int Port,
    [property: JsonPropertyName("seen_at")]   long SeenAt,
    [property: JsonPropertyName("reachable")] bool? Reachable
);

public record SyncStatus(
    [property: JsonPropertyName("peers")]        IReadOnlyList<SyncPeer> Peers,
    [property: JsonPropertyName("last_sync_at")] long LastSyncAt,
    [property: JsonPropertyName("device_id")]    string DeviceId
);

public record VaultStatus(
    [property: JsonPropertyName("initialized")] bool Initialized,
    [property: JsonPropertyName("unlocked")]    bool Unlocked,
    [property: JsonPropertyName("status")]      string Status
);
